File: src/main/java/io/gittul/domain/user/UserController.java
package io.gittul.domain.user;

import io.gittul.domain.follow.FollowService;
import io.gittul.domain.user.dto.UserDetailResponse;
import io.gittul.domain.user.entity.User;
import io.gittul.infra.auth.aop.Authenticated;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/users")
@RequiredArgsConstructor
public class UserController {

    private final UserService userService;
    private final FollowService followService;

    @PostMapping("/follow")
    public void follow(@Authenticated User user, @RequestBody Long followeeId) {
        followService.follow(user, followeeId);
    }

    @PostMapping("/unfollow")
    public void unfollow(@Authenticated User user, @RequestBody Long followeeId) {
        followService.unfollow(user, followeeId);
    }

    @PostMapping("interest")
    public void interest(@Authenticated User user, @RequestBody List<String> tags) {
        userService.interest(user, tags);
    }

    @PostMapping("uninterest")
    public void uninterested(@Authenticated User user, @RequestBody String tag) {
        userService.unInterest(user, tag);
    }

    @GetMapping("/me")
    public UserDetailResponse me(@Authenticated User user) {
        return UserDetailResponse.of(user);
    }
}

File: app/src/main/kotlin/io/gittul/app/domain/user/UserController.kt
package io.gittul.app.domain.user

import io.gittul.app.domain.follow.FollowService
import io.gittul.app.domain.user.dto.UserDetailResponse
import io.gittul.app.infra.auth.aop.AccessGuard
import io.gittul.app.infra.auth.aop.AuthContext
import org.springframework.web.bind.annotation.*

@AccessGuard
@RestController
@RequestMapping("/users")
class UserController(
    private val userService: UserService,
    private val followService: FollowService
) {

    @PostMapping("/follow")
    fun follow(
        @RequestBody followeeId: Long
    ) {
        val user = AuthContext.getUser();
        followService.follow(user, followeeId)
    }

    @PostMapping("/unfollow")
    fun unfollow(
        @RequestBody followeeId: Long
    ) {
        val user = AuthContext.getUser();
        followService.unfollow(user, followeeId)
    }

    @PostMapping("interest")
    fun interest(
        @RequestBody tags: List<String>
    ) {
        val user = AuthContext.getUser();
        userService.interest(user, tags)
    }

    @PostMapping("uninterest")
    fun uninterested(
        @RequestBody tag: String
    ) {
        val user = AuthContext.getUser();
        userService.unInterest(user, tag)
    }

    @GetMapping("/me")
    fun me(): UserDetailResponse {
        val user = AuthContext.getUser();
        return UserDetailResponse.of(user)
    }
}
